fastlane.md

```diff
@@ -0,0 +1,170 @@
+    It could be as simple as a `lane` that calls `upload_to_play_store`. The
+    `apk` argument can be set to `../build/app/outputs/apk/release/app-release.apk`
+    to use the apk `flutter build` already built.
+    * ![iOS](/images/fastlane-cd/ios.png) On iOS, follow the [Fastlane iOS beta deployment guide](https://docs.fastlane.tools/getting-started/ios/beta-deployment/).
+    It could be as simple as a `lane` that calls `build_ios_app` with
+    `export_method: 'app-store'` and `upload_to_testflight`. An extra build is
+    used after `flutter build` on iOS since `flutter build` builds and .app and
+    doesn't archive .ipas for release.
+
+You're now ready to perform deployments locally or migrate the deployment
+process to a CI system.
+
+## Running deployment locally
+
+1. Build the release mode app.
+    * ![Android](/images/fastlane-cd/android.png) `flutter build apk --release`.
+    * ![iOS](/images/fastlane-cd/ios.png) `flutter build ios --release --no-codesign`.
+    No need to sign now since we'll archive using Fastlane which will sign at that point.
+1. Run the Fastfile script on each platform.
+    * ![Android](/images/fastlane-cd/android.png) `cd android` then
+    `fastlane [name of the lane you created]`.
+    * ![iOS](/images/fastlane-cd/ios.png) `cd ios` then
+    `fastlane [name of the lane you created]`.
+
+## Cloud build and deploy setup
+
+Follow the local setup section above first to make sure the process works before
+migrating onto a cloud system like Travis.
+
+The main aspects to consider is that since cloud instances are ephemeral and
+untrusted, you won't be leaving your credentials like your Play Store service
+account JSON or your iTunes distribution certificate on the server.
+
+Both [Travis](https://docs.travis-ci.com/user/environment-variables/#Encrypting-environment-variables)
+and [Cirrus](https://cirrus-ci.org/guide/writing-tasks/#encrypted-variables)
+(randomly chosen examples, you can use any CI systems you wish) support the
+concept of encrypted environment variables to store private data.
+
+**Take precaution to not re-echo those variable values back onto the console in
+your test scripts**. Those variables are also not available in pull requests
+until they're merged to ensure that malicious actors cannot create a pull
+request that prints these secrets out. Be careful with interactions with these
+secrets in pull requests you accept and merge.
+
+1. Make login credentials ephemeral.
+    * ![Android](/images/fastlane-cd/android.png) On Android:
+        * Remove the `json_key_file` field from `Appfile` and store the string
+        content of the JSON in your CI system's encrypted variable. Use the
+        `json_key_data` argument in `upload_to_play_store` to read the
+        environment variable directly in your `Fastfile`.
+        * Serialize your upload key such as with base64 and save as a encrypted
```

Comment:
Serialize your upload key (for example, using base64), and save it as...

Reply:
done